app/src/main/cpp/skyline/common.h

```diff
@@ -338,6 +339,35 @@ namespace skyline {
         constexpr std::size_t Hash(std::string_view view) {
             return frz::elsa<frz::string>{}(frz::string(view.data(), view.size()), 0);
         }
+
+        /**
+         * @brief Returns the most appropriate integral type and count to represent the given T object
+         */
+        template<class T>
+        struct IntegralFor {
+            typedef typename std::conditional_t<sizeof(T) % sizeof(u64) == 0, u64,
+                typename std::conditional_t<sizeof(T) % sizeof(u32) == 0, u32,
+                    typename std::conditional_t<sizeof(T) % sizeof(u16) == 0, u16, u8>
+                >
+            > type;
+
+            static constexpr size_t count{sizeof(T) / sizeof(type)};
+        };
+
+        namespace detail {
+            static thread_local std::mt19937_64 generator{GetTimeTicks()};
+        }
+
+        /**
+         * @brief Fills an array with random data from a Mersenne Twister pseudo-random generator
+         * @note The generator is seeded with the the current time in ticks
+         */
+        template<typename T> requires (std::is_integral_v<T>)
+        void FillRandomBytes(std::span<T> in) {
+            std::independent_bits_engine<std::mt19937_64, std::numeric_limits<T>::digits, T> gen(detail::generator);
+
+            std::generate(in.begin(), in.end(), gen);
+        }
```

Comment:
Do one that is `requires (!std::is_integral_v<T>)` with just `void FillRandomBytes(T object)` and calls `FillRandomBytes(std::span(reinterpret_cast<PointerFor<T>::type>(&object), PointerFor<T>::count))` which should handle the case for non-integral types then utilize this for the UUIDs. Alternatively, you can make a `T CreateRandomObject()` which takes nothing but returns an object filled with random bytes of whatever the supplied template type is.